src/main/java/Deadline.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Class for the deadline task.
+ * Contains a description of the task and the datetime for which it is due.
+ */
+public class Deadline extends Task {
+
+    protected String by;
+
+    /**
+     * Constructor for Deadline task.
+     * Initializes the description and due datetime.
+     *
+     * @param des description of the deadline.
+     * @param by datetime for which the deadline is due.
+     */
+    public Deadline(String des, String by) {
+        super(des);
+        this.by = by;
```

Comment:
"this" keyword used for your tasks constructors, but some methods not used. 
Good to be consistent with the usage!